job/src/slipstream/job/job.py

```diff
@@ -7,6 +7,12 @@
 from .util import classlogger
 
 
+class NonexistentJobError(Exception):
```

Comment:
This was done to avoid having the SlipStream API leak directly into the Job class.  If you think it is cleaner, the check could be moved there.